### ISSUE TITLE
💅 useSemanticElements not fired on self-closing tags

### ISSUE THREAD

**Issue by sterliakov** [biomejs/biome#6622]
### Environment information

```bash
CLI:
  Version:                      2.0.6
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.19.3
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0
```

### Rule name

lint/a11y/useSemanticElements

### Playground link

https://biomejs.dev/playground/?indentStyle=space&quoteStyle=single&lintRules=all&enabledAssist=false&allowComments=false&code=dAB5AHAAZQAgAFIAZQBhAGMAdABOAG8AZABlACAAPQAgAHUAbgBrAG4AbwB3AG4AOwAKAAoAZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABCAGEAZAAoACkAOgAgAFIAZQBhAGMAdABOAG8AZABlACAAewAKACAAIAByAGUAdAB1AHIAbgAgACgACgAgACAAIAAgADwAcwBwAGEAbgAKACAAIAAgACAAIAAgAHIAbwBsAGUAPQAiAGIAdQB0AHQAbwBuACIACgAgACAAIAAgACAAIAB0AGEAYgBJAG4AZABlAHgAPQB7ADAAfQAKACAAIAAgACAALwA%2BAAoAIAAgACkACgB9AAoA

### Expected result

Diagnostic should be issued for both spans - they are exactly equivalent.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by sterliakov** [comment]
(for the record, the inspiring [eslint rule in jsx-a11y](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/prefer-tag-over-role.md) does fire on both)

**Comment by zk2k2** [comment]
Hello I would like to work on this issue, please assign it to me!

**Comment by ematipico** [comment]
I am going to unassign @zk2k2 because it's been a month without updates
